doc/lingy.md

```diff
@@ -415,8 +433,80 @@ Certainly there are other differences and this section will be improved over
 time.
 
 
+# Hacking on Lingy
+
+If you want to contribute to Lingy, Welcome!
+
+If you just want to play with the source while using Lingy, Welcome!
+
+Here, we'll cover how to install Lingy from source and modify the source code.
+This will include tips on debugging Lingy.
+
+This section is a work in progress...
+
+Lingy development requires `git`, `bash`, `perl`, `make`, `curl` (almost any
+versions of these) and a bunch of Perl modules.
+
+You probably already have the commands.
+The best way to install the dependency modules is using the `cpanm Lingy`
+command described above in the "Installation" section.
+The will install Lingy and all the Perl dependencies.
+
+Then you can `git clone` the source code:
+```
+git clone https://github.com/ingydotnet/lingy.git
+```
+
+Set up local environment variables for development:
+```
+export PATH=$PWD/lingy/perl/bin:$PATH
+export PERL5LIB=$PWD/lingy/perl/lib
+```
+
+And test your local source code install of Lingy:
+```
+which lingy     # Should point at your cloned version
+lingy --execs   # Should list your clone
+lingy -e '(println *lingy-version*)'
+```
+
+Add the `export`s above to your shell rc file if you want to.
+
+You might need to add these exports if Lingy warns about them.
+This is more common for people who haven't used Perl in a while since people
+who do likely took care of it in the past.
+```
+LC_CTYPE=en_US.UTF-8
+LC_ALL=en_US.UTF-8
+```
+
+
+## Debugging Lingy in the REPL
```

Comment:
Not for this particular section, but with the rest of Lingy docs I think it is important that we don't confuse **the REPL** with the **the REPL prompt**.

* The REPL is a process responsible for interactively reading, updating and inspecting the application under development.
* The REPL prompt is a UI to this process.

In Clojure it is mostly recommended that the goto-UI for the REPL is the files with the application source code. And Clojure development environments provide integration for making this convenient and delightful. See https://youtu.be/d0K1oaFGvuQ for an example of the workflow. The shortest path to enabling this is to provide an nREPL server, which will make it easy to use any existing Clojure IDE with Lingy.